go.mod

```diff
@@ -71,3 +72,5 @@ replace github.com/evanphx/json-patch => github.com/evanphx/json-patch v4.9.0+in
 
 // pkg disappeard from bitbucket
 replace bitbucket.org/ww/goautoneg => github.com/munnerz/goautoneg v0.0.0-20120707110453-a547fc61f48d
+
+replace github.com/operator-framework/operator-sdk => github.com/luksa/operator-sdk v0.0.0-20210824135520-3b25565223d4
```

Comment:
We're using a _very_ old version. I don't think this code is even in operator-sdk anymore. This is just a quick fix until we upgrade.

Reply:
Yeah, my opinion is that we submit this upstream, get it merged there and we will get it on the next rebase, in 2.2.

Reply:
I'm saying we can't submit this upstream, because the code that I've fixed has been removed completely. They won't accept PRs for the ancient version of the library that we're using. In 2.2 we'll use an up-to-date version and there won't be any more unnecessary API discovery calls. But since the changes I've made are a considerable improvement (and they are simple enough to not introduce any new issues) I think we should get them into 2.1.

Reply:
I got it Marko. I understand all you are saying. My point is: Do we really need this for 2.1, especially at this point where we are working on stabilizing? If the team agrees that we do need this, then at least we should for the repo under maistra and open a pr there and use this fork here.

Or, we could wait for 2.2 when we can bump to the latest sdk (my vote).

Reply:
The existing code performs full API discovery for each of the operator's custom resources and also a full API discovery simply to determine whether one specific resource type exists. This is such a bad implementation that I'd almost consider it a bug. If the fix wasn't as simple as it is, I would be fine with waiting for 2.2, but since the fix is really simple (perform a single API discovery call and then use the retrieved data when processing all of the custom resources), I don't see why we wouldn't include it in 2.1 The work is finished and there's no additional work that QE needs to do. 

Sure I could fork the repo under maistra, but since we're vendoring dependencies anyway (and we already do vendor in third party repos that are owned by individuals and not organizations) I didn't think it was that important to fork it.